### ISSUE TITLE
Oct 26 - 30 - Take Your First Steps with C#

### ISSUE THREAD

**Issue by mebaumb** [microsoft/ReactorPlanning#113]
# Event Information
| Category | Details |
|-----------|---------|
| Reactor Topic | Coding, Languages, and Frameworks |
| Monthly Topic (if applicable) | Open Source |
| Weekly Topic (if applicable) |  C# |
| Workshop Duration | 60 Mins | 
| Title | Take Your First Steps with C# |
| Description |<p>The C# programming language offers great productivity, versatility, and is the most powerful programming language for the .NET Framework. If you are interested in becoming a C# programmer, this workshop is a great first step. You’ll start learning the basics of the syntax by writing your first C# code. You’ll then explore how to store and retrieve data in applications and how to perform basic data formatting. If you want to follow along, we'll be going through the content in this Microsoft Learn path: https://aka.ms/IntrotoCSharp-2. You will leave this workshop with an increased knowledge about programming in C# and resources to help you continue your journey.</p><p>Prerequisites: Prior programming experience is helpful but not required. </p>|
| Learn Content | https://docs.microsoft.com/learn/paths/csharp-first-steps/?WT.mc_id=reactor-github-reactor (modules 1-3) |

# Workshop(s) details
| Date |  Local Start Time |  Location |  Speaker | SharePoint ID | Platform |
|------|-------------------|-----------|----------|---------------|----------|
| Oct 27 | 6:30 PM | Bangalore | TBD | 11646 | TBD |
| Oct 27 | 4:00 PM PST | US/Canada/Sydney | Will Goode (Chris Bautista - Moderator) | 11647 | TBD |
| Oct 28 | 7:30 PM | Sao Paulo | TBD | 11648 | TBD |
| Oct 28 | 7:30 PM | Shanghai | TBD | 11649 | TBD |
| Oct 28 | 5:00 PM | London, Stockholm, Tel Aviv, Abu Dhabi | TBD | 11650 | TBD |

# Event Checklist

## 10 Weeks Prior
- [x] Reactor Content Team (RCT) opens a new issue for an upcoming local workshop filling in the Event Information and Workshop details tables. RCT member who owns content (content owner) this week will be the assignee.
- [x] RCT and content owner discuss details of the event in the comments of this issue. RCT adds a _confirmed-event_ label and _local-digital-event_ tag to the issue and adds event to the __Local Digital Events__ project. Issue will appear in the __Upcoming Streams__ column
- [x] RCT adds a _ready-for-marketing-edit_ label so the RCT can review title and description

## 8 Weeks Prior
- [x] RCT does a review of title and description and removes the _ready-for-marketing-edit_ label 
- [x] RCT moves the issue to the __Ready for SharePoint__ column
- [ ] RCT informs the Reactor Marketing Team of the new event information
- [ ] RCT confirms the speaker
- [ ] RCT provides speaker with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT moves the issue to the __Content Preparation In Progress__ column

## 5 weeks Prior
- [ ] Content owner opens a PR with the new content for the workshop on the [customer-facing Reactor repository](https://github.com/microsoft/Reactors/tree/main/online-event-resources) and adds the _content-development_ and _local-digital-event_ tag to the PR. 
- [ ] RCT reviews, approves, and merges the changes
- [ ] RCT shares content with the vendor/instructor
- [ ] RCT provides vendor/instructor with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT ensures that instructors have received an invite to the workshops and that PM team works with instructors to get setup in the platform (e.g. Teams, Teams Live Event)

## 1 Week Prior
- [ ] Connect vendor/instructor with the PM(s) for the locations where this event has been marketed
- [ ] Check-in with vendor/instructor that they are prepared for the workshop
- [ ] RCT moves the issue to the __Ready to Stream__ column

## During and After Stream
- [ ] Instructor (and optional moderator) run the stream
- [ ] _(Optional)_RCT submits recording to be uploaded to the Reactor YouTube and to be made available on LearnTV
- [ ] RCT moves issue to the __Stream Complete__ column

**Comment by mebaumb** [comment]
Details are in Dynamics, Need to update it with the final

**Comment by meaghanlewis** [comment]
Here's an overview of what this workshop will cover and suggested agenda to follow with links and resources.

## Content to teach
This workshop will take Reactor community members through the first three modules of
- [Take your first steps with C#](https://aka.ms/FirstStepsinCSharp-5)

**This link should be shared with attendees during the workshop:**
- https://aka.ms/FirstStepsinCSharp-8

Encourage attendees to go through the modules with you and complete each one so that they get acheivement badges in their profile.

## Steps
1. [Write your first C# code](https://docs.microsoft.com/en-us/learn/modules/csharp-write-first/)
    - Show how to install
    - Share C# guide: https://docs.microsoft.com/dotnet/csharp/
    - C# extension for VS Code: https://marketplace.visualstudio.com/items?itemName=ms-dotnettools.csharp
2. [Store and retrieve data using literal and variable values in C#](https://docs.microsoft.com/en-us/learn/modules/csharp-literals-variables/)
3. [Perform basic string formatting in C#](https://docs.microsoft.com/en-us/learn/modules/csharp-basic-formatting/)
4. Next steps
    - Encourage attendees to complete the other modules in the path: https://aka.ms/FirstStepsinCSharp-8
4. Closing
    - Q&A
    - Survey: https://aka.ms/Reactor/Survey
      - Note: The event code is listed in this issue description and/or your Reactor contact can share that with the community members
    - Share Resources
    - Check if the Reactor contact has other information to share

**Comment by KimberlyBolton1** [comment]
@meaghanlewis edited abstract for your review:

**Take Your First Steps with C#**
The C# programming language offers great productivity, versatility, and is the most powerful programming language for the .NET Framework. If you are interested in becoming a C# programmer, this workshop is a great first step. You’ll start learning the basics of the syntax by writing your first C# code. You’ll then explore how to store and retrieve data in applications and how to perform basic data formatting. You will leave this workshop with an increased knowledge about programming in C# and resources to help you continue your journey.
Prerequisites: This workshop assumes you have a basic understanding knowledge of a programming language other than Python. To follow along, install Visual Studio and Visual Studio Code at (https://visualstudio.microsoft.com/downloads/).

**Comment by meaghanlewis** [comment]
Thanks @KimberlyBolton1!

For the prerequisites, I don't think a basic understanding of a programming language is necessary, although it could be helpful. I also wouldn't call out Python in particular. And I'm just realizing that the Learn content uses an in-browser .NET Editor, so neither Visual Studio or VS Code are needed.

**Comment by KimberlyBolton1** [comment]
@meaghanlewis thanks for the feedback. The Python was a cut and paste mistake on my part. How about this:

Take Your First Steps with C#
The C# programming language offers great productivity, versatility, and is the most powerful programming language for the .NET Framework. If you are interested in becoming a C# programmer, this workshop is a great first step. You’ll start learning the basics of the syntax by writing your first C# code. You’ll then explore how to store and retrieve data in applications and how to perform basic data formatting. You will leave this workshop with an increased knowledge about programming in C# and resources to help you continue your journey.

Prerequisites: Prior programming experience is helpful but not required.

**Comment by meaghanlewis** [comment]
@KimberlyBolton1 looks great! ✅

**Comment by KimberlyBolton1** [comment]
@mebaumb abstract file and moved to Ready for Sharepoint.

**Comment by mebaumb** [comment]
Title/abstract sent to Caio for translation.

US/Canada/Syndey published

Waiting on Speaker confirmation to publish the other ones!

**Comment by mebaumb** [comment]
Bangalore, Sao Paulo, Shanghai and London/Stockholm/Tel Aviv/Abu Dhabi published without speaker info (9/2)

**Comment by mebaumb** [comment]
Added the learn link to the description on Dynamics/Meetup (10/23)

**Comment by mebaumb** [comment]
These events have been completed and attendance data has been added to Dynamics.
